roles/openshift_health_checker/library/search_journalctl.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+"""Interface to journalctl"""
+
+from time import time
+import json
+import re
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+class InvalidMatcherRegexpException(Exception):
+    """Base exception class for invalid matcher regexp fields"""
+    pass
+
+
+class InvalidMatcherException(Exception):
+    """Base exception class for missing or invalid matcher fields."""
+    def __init__(self, message, missing_fields=None):
+        super(InvalidMatcherException, self).__init__(message)
+        self.missing_fields = missing_fields
+
+
+def main():
+    """Scan a given list of matchers for journalctl messages containing given patterns."""
+    module = AnsibleModule(
+        argument_spec=dict(
+            log_count_limit=dict(type="int", default=500),
+            log_matchers=dict(type="list", required=True),
+        ),
+    )
+
+    timestamp_limit_seconds = time() - 60 * 60  # 1 hour
+
+    log_count_limit = module.params["log_count_limit"]
+    log_matchers = module.params["log_matchers"]
+
+    matched_regexp, errors = get_log_matches(log_matchers, log_count_limit, timestamp_limit_seconds)
+
+    module.exit_json(
+        changed=False,
+        failed=len(errors),
+        errors=errors,
+        matched=matched_regexp,
+    )
+
+
+def get_log_matches(matchers, log_count_limit, timestamp_limit_seconds):
+    """Iterate through a list of matchers, verify required fields exist on each one, and
+    retrieve log output for each matcher, appending its match pattern to a collection if
+    a matcher's regex pattern is found.
+    Return a collection of matched regular expressions, and a list of errors."""
+    matched_regexp = []
+    errors = []
+
+    for matcher in matchers:
+        try:
+            ensure_matcher_fields(matcher)
+        except InvalidMatcherException as err:
+            errors.append(str(err))
+            continue
+
+        log_input, error = get_log_input(matcher)
```

Comment:
This seems inconsistent, why do we use exceptions in `ensure_matcher_fields` but use a return value for errors in `get_log_input`?!